tests/subsys/modbus/src/test_modbus_server.c

```diff
@@ -183,13 +183,17 @@ static struct modbus_iface_param server_param = {
 	},
 };
 
+#if DT_NODE_EXISTS(DT_INST(1, zephyr_modbus_serial))
```

Comment:
Can you please justify the use of a hard-coded instance 1 here?

Reply:
For testing on the hardware we need two interfaces, one is client (inst 0) and the other is server (inst 1), see harness config. If the board does not support two interfaces, the test is build-only.

Reply:
there's no guarantee 0 will be client and 1 server, right?

Reply:
IIRC yes. But It does not matter who is who. They just have to be different interfaces.

Reply:
Please add some source code comments explaining this. Hard-coded instance numbers -- especially nonzero values -- always deserve careful consideration, so let's record the reasoning in the code to avoid future confusion.